### ISSUE TITLE
Error connecting to db-mssql

### ISSUE THREAD

**Issue by vaqxai** [tpope/vim-dadbod#175]
I added my integrated security database with the url:
`sqlserver://db-mssql`
If I were connecting with sqlcmd, i'd go:
`sqlcmd -S db-mssql -E -d mydatabase`
which works correctly.

But, when trying the connection url listed above, and as you can see I am omitting the `user` property, which should mean (I don't know vimscript well enough to tell from the adapter code) that there won't be a `-U` or `-P` included in the sqlcmd command, I get the following error from dadbod:

`Error connecting to db db-mssql: DB exec error: Sqlcmd: The -E and the -U/-P options are mutually exclusive`

What can I do to get the plugin to not use `-U/-P` and only provide the `-E` option?

**Comment by tpope** [comment]
The SQL Server delegates to the `#as_argv()` helper to decide whether to add `-U`: https://github.com/tpope/vim-dadbod/blob/37523ffe0849b51db07fa7f48430c2aca5420a0e/autoload/db/url.vim#L130-L133

This code is pretty straightforward: no user, no `-U`. So very unclear how this can happen.

You can check the exact command with the following command:

 ```vim
:echo db#adapter#sqlserver#interactive(db#resolve('sqlserver://db-mssql'))
```

Perhaps seeing the exact command line will offer a clue how it's being assembled.

**Comment by vaqxai** [comment]
The result of
`:echo db#adapter#sqlserver#interactive(db#resolve('sqlserver://db-mssql'))`
is `['sqlcmd', '-S', 'db-mssql', '-E']`.
I have no idea where the error comes from. The command looks to be resolved properly. Running `sqlcmd -S db-mssql -E` in terminal yields a proper connection and ability to execute commands.

I'm on Windows if that matters

**Comment by tpope** [comment]
Perhaps it's getting confused by an environment variable which is only set in Vim?

**Comment by tpope** [comment]
In the `s:job_run` function. Neovim uses `jobstart()` and Vim uses `job_start()`.

**Comment by vaqxai** [comment]
So, the command it runs exactly is:
`sqlcmd -S db-mssql -E -1 C:/Users/Vaqxai/AppData/Local/Temp/nvim.0/K1MQyj/1`
And running that command in the terminal results in:
`Sqlcmd: The -E and the -U/-P options are mutually exclusive.`
The folder `K1MQyj` is empty, there is no `1` inside (folder or file), but including the `-1` option, which i am not sure what it does, perhaps redirects the interactive session to a file so dadbod can read/write to it, causes the error to appear.

**Comment by tpope** [comment]
The `-i` comes from here: https://github.com/tpope/vim-dadbod/blob/7888cb7164d69783d3dce4e0283decd26b82538b/autoload/db/adapter/sqlserver.vim#L56

No idea what could change it to `-I`, but that's definitely a problem.

**Comment by vaqxai** [comment]
After today's testing I found that it doesn't get changed to `-I` (anymore?), however, by checking the command at each step what I found it executes (now) is exactly: `sqlcmd -S db-mssql -E -i C:/Users/Vaqxai/AppData/Local/Temp/nvim.0/Blf48x/3`

So, naturally, seeing that it's `-i` now, I ran it in terminal, annnd:
```
$ sqlcmd -S localhost -E -d db-mssql -i C:/Users/Vaqxai/AppData/Local/Temp/nvim.0/Blf48x/3
Sqlcmd: The -E and the -U/-P options are mutually exclusive.
```

First idea, change the direction of the slashes from `/` to `\`, result:
```
$ sqlcmd -S localhost -E -d db-mssql -i C:\Users\Vaqxai\AppData\Local\Temp\nvim.0\Blf48x/3
Sqlcmd: 'C:\Users\Vaqxai\AppData\Local\Temp\nvim.0\Blf48x\1': Invalid filename.
```

Bingo. Now where does the path come from, then?
`tempname()`, `:echo tempname()` returns the path with incorrect slash direction for Windows.
Why does it do that? Because I set `shellslash` in my config. Why did I do that? Because i was using  `fish` which was on cygwin and required unix-like paths, and now I'm using powershell which doesn't.

**Closed as user being dumb.**
For future reference: If on windows, check shellslash
(that might also explain why other stuff was wonky-breaky)
